### DISCUSSION TITLE
MacOS Keyboard/Input Issues

### DISCUSSION THREAD

**Discussion by sean64** [logisim-evolution/logisim-evolution#1366] [Q&A] [answered]
Wondering if anyone has come across this and if so, is there a workaround?
After starting logisim-evolution everything works perfectly, after using it for a short time, sometimes, the keyboard stops working.  I can't use shortcuts and I can't type anything in label fields, etc.
If I restart the program or just keep using my pc, the problem goes away... sometimes...  I can't seem to find any pattern of behavior that causes this yet and am still trying to troubleshoot it.
Sometimes it works fine without issue, but more often than not, it interrupts my workflow.
Any ideas?

**Comment by maehne** [comment]
I have not yet observed such behaviour on macOS. However, I am still stuck on macOS 10.13. Please, be more specific about which version of macOS and Logisim-evolution you are using. Did you try the latest stable version 3.7.2? Did you use the DMG released on this GitHub page or did you built it yourself? In that case, it is important to know which version of OpenJDK did you use.

**Comment by sean64** [comment, marked as answer]
Thanks for the response.  I've spent the weekend trying to figure this out and I may have the answer.
I am using macOS 12.1 with the latest version of the pre-built logisim-evolution.
I am also using Synergy with a PC as the server and the Mac as a client.  It appears to be related to that.  I used a directly connected mouse over the weekend and had no issues at all.
Oddly, I was always using a directly connected keyboard.  For some reason Synergy's mouse sharing was screwing up my keyboard input... or so it seems at this point.
Thanks for your attention!

**Reply by MarcinOrlowski** [reply]
Please do observe this for a few more days and if you still believe this is the culprit that caused our problems, please mark your post as answer to close the thread.

**Reply by sean64** [reply]
Thanks for the reminder.
Since deactivating Synergy, I have spent the last 2 days with no issues.  I've not been able to use it more than an hour without issue in the past, pretty sure that was the issue.

Thanks again!
